java/com/google/javascript/jscomp/CheckStrictDeps.java

```diff
@@ -149,7 +149,7 @@ private void checkNamespaceIsProvided(NodeTraversal t, Node n, String namespace)
           namespace += ".js";
         }
 
-        Webpath me = Webpath.get(t.getSourceName());
+        Webpath me = Webpath.get(convertPathToModuleName(t.getSourceName(), state.roots).or(t.getSourceName()));
```

Comment:
What does convertPathToModuleName return in practice for "./genlib" ? 

Looking at the implementation, it appears to throw an IllegalArgumentException if the import starts with '/', which is the recommended way to import things. Do all the existing tests pass? 

Similarly, I recommend extensionless imports, so that imports for jsx and js work equivalently.

Reply:
`t.getSourceName()` is the name of the importing file not the actual import string. I believe it's safe to use in this way since it's also done on L#90 / L#94. Here's a table of the different values for the different cases (extension vs extensionless doesn't affect these).

| Case                      | original ns        | t.getSourceName()                                 | converted module name | me                  | resolved ns        |
|---------------------------|--------------------|---------------------------------------------------|-----------------------|---------------------|--------------------|
| absolute raw src | /js/demo/lib.js    | js/demo/main.js                                   | /js/demo/main.js      | /js/demo/main.js    | /js/demo/lib.js    |
| relative raw src | ./lib.js           | js/demo/main.js                                   | /js/demo/main.js      | /js/demo/main.js    | /js/demo/lib.js    |
| absolute gen src | /js/demo/libgen.js | bazel-out/darwin-fastbuild/bin/js/demo/maingen.js | /js/demo/maingen.js   | /js/demo/maingen.js | /js/demo/libgen.js |
| relative gen src        | ./libgen.js        | bazel-out/darwin-fastbuild/bin/js/demo/maingen.js | /js/demo/maingen.js   | /js/demo/maingen.js | /js/demo/libgen.js |

Prior to this change, in the relative gen src case it looks like:

| Case             | original ns | t.getSourceName()                                 | converted module name | me                                                 | resolved ns                                       |
|------------------|-------------|---------------------------------------------------|-----------------------|----------------------------------------------------|---------------------------------------------------|
| relative gen src | ./libgen.js | bazel-out/darwin-fastbuild/bin/js/demo/maingen.js | /js/demo/maingen.js   | /bazel-out/darwin-fastbuild/bin/js/demo/maingen.js | /bazel-out/darwin-fastbuild/bin/js/demo/libgen.js |

The existing tests do pass.

Reply:
Ahhh, thanks for clarifying!